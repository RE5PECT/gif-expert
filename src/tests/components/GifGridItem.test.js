import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en componente GifGridItem', () => {
  const title = 'Un titulo';
  const url = 'https://localhost/prueba';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should show the component GifGridItem', () => {  
    expect(wrapper).toMatchSnapshot();
  });

  test('should tener un parrafo con el title', () => {
   const p = wrapper.find('p');
   expect(p.text().trim()).toBe(title);
  });

  test('should tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
   });

   test('should have the class animate__bounceInRight', () => {
    const div = wrapper.find('div').at(0);
    expect(div.hasClass('animate__bounceInRight')).toBe(true);
   }); 
});
