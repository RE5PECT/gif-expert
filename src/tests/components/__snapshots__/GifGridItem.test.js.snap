// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGridItem should show the component GifGridItem 1`] = `
<div
  className="card animate__animated animate__bounceInRight"
>
  <img
    alt="Un titulo"
    src="https://localhost/prueba"
  />
  <p>
    Un titulo
  </p>
</div>
`;
